docs/developer/thumbnails.mdx

```diff
@@ -0,0 +1,100 @@
+---
+title: Thumbnails
+---
+
+## Introduction
+
+This guide describes how to request for a thumbnail and how the thumbnails are generated.
+
+### Thumbnails sizes and formats
+
+Thumbnails are available on the following models and fields:
+
+- `User.avatar`
+- `OrderLine.thumbnail`
+- `Product.thumbnail`
+- `Collection.background_image`
+- `Category.background_image`
+- `ProductMedia.url`
+- `ProductImage.url`
+
+You can specify the `size` and `format` of the thumbnail. You are free to pass any integer
+as `size`, but what ends up being used is the closest value from the following list:
+[32, 64, 128, 256, 512, 1024, 2048, 4096].
+For example: when you pass "55", we resolve it to "64".
+
+Currently, `WebP` is the only supported value for `format`.
+Both `size` and `format` fields are optional, but when you decide to provide `format`,
+the `size` becomes required as well. If either, or both of these are not specified,
+the original image attributes will be used.
+
+### Generating the thumbnails
+
+The thumbnails are generated on-demand. When the user requests the thumbnail with the
+given size and format for the first time, the proxy URL is returned. When the proxy URL
+is opened, the requested thumbnail is created and the user gets the requested image.
+In case of a second and further request for the same thumbnail, the previously created
+thumbnail's URL is returned.
+
+### Example
+
+Here is an example query for the category's background image of `size` = "100" and `format` = "WebP":
+
+```graphql
+query {
+  category(id: "Q2F0ZWdvcnk6MjU=") {
+    name
+    backgroundImage(size: 100, format: WEBP) {
+      url
+      alt
+    }
+  }
+}
+```
+
+As the query ran for the first time, the response contains proxy URL:
```

Comment:
You mean: "After the query runs ..., it contains the proxy?"
"As" introduces two actions happening at the same time, so here it would mean that the response contains URL **in the middle of** running the query. What we want to say is: The query runs for the first time and as a result the response contains proxy URL, right? Or am I wrong?

Reply:
That's correct. It was more like "because the query ran for the first time, the response looks like this. It would look differently, if it would run again"